[WIP] doc: add v0.10.0 changelog

kolyshkin opened the pull request:
Not finalized but good enough for a review.

pcmoore commented:
I don't feel strongly on this either way, but I've always felt manually written CHANGELOGs should be more than just the subject lines of relevant patches.  For example, I would take the following lines:
```
- doc: Update CONTRIBUTING.md.
- doc: Update README to support markdown.
- doc: dedup README.md and CONTRIBUTING.md.
- doc: add admin docs and SECURITY.md.
- doc: README: rm old go/gcc version warning.
- doc: remove reference to the defunct mailing list.
```
... and write them like this:
```
- Updated the main docs and converted to README.md
- Added CONTRIBUTING.md, SECURITY.md, and administrative docs under doc/admin
```

pcmoore commented:
I also wouldn't worry about periods at the end of the changelog entries.  In some ways it would be more consistent with the existing changelog to *not* use periods.

pcmoore reviewed (approved):
Regardless, neither of the above are significant in any way, so if you like it the way it is, and @drakenclimber is okay with it, I am too.
```
Acked-by: Paul Moore <paul@paul-moore.com>
```

drakenclimber commented:
I don't have a strong opinion one way or the other, but I am a fan of consistency.  Thanks for all the help, @kolyshkin.

`Acked-by: Tom Hromatka <tom.hromatka@oracle.com>`

kolyshkin commented:
> I've always felt manually written CHANGELOGs should be more than just the subject lines of relevant patches

That is what I aimed for. There are quite a few edits (for clarity), merges (e.g. about KillThread/KillProcess), and removals (like about adding travis CI which was later removed). Yet apparently I could have done more.

> ```
> - Updated the main docs and converted to README.md
> - Added CONTRIBUTING.md, SECURITY.md, and administrative docs under doc/admin
> ```

Updated.

Similar changes (compacted and merged) to a few other lines.

>  I also wouldn't worry about periods at the end of the changelog entries. In some ways it would be more consistent with the existing changelog to not use periods.

Periods removed.

pcmoore commented:
Thanks @kolyshkin
